tests/helpers.sh

```diff
@@ -93,23 +93,21 @@ fi
 selfsign_cert() {
     KEY="$1"
     export CERT="$2"
-    shift 2
+    export SUBJ="$3"
+    shift 3
     export AARGS="$*"
 
-    TMPCA=${TMPPDIR}/tmpca
-    TMP_OPENSSL_CONF=${OPENSSL_CONF}
-    sed -e "s|^dir .*|dir = ${TMPCA}|" \
-        "${OPENSSL_CONF}" > "${OPENSSL_CONF}.tmpca"
-    export OPENSSL_CONF=${OPENSSL_CONF}.tmpca
-
-    mkdir -p "${TMPCA}/newcerts" "${TMPCA}/private"
-    if [ ! -e "${TMPCA}/serial" ]; then
-        echo "01" > "${TMPCA}/serial"
-    fi
-    touch "${TMPCA}/index.txt"
-
     title PARA "Generating a new selfsigned certificate for ${KEY}"
-    ossl 'req -batch -noenc -x509 -new -key ${KEY} ${AARGS} -out ${CERT}'
+    ossl 'x509 -new -subj /CN=${SUBJ}/ -days 365 -extensions v3_ca -extfile ${OPENSSL_CONF} -out ${CERT} -outform PEM -signkey ${KEY} ${AARGS}'
     echo "$helper_output"
-    OPENSSL_CONF=${TMP_OPENSSL_CONF}
+}
+
+ptool() {
+    # Unset the openssl.cnf here to prevent pkcs11-tool from attempting to load
+    # pkcs11.so, the loading itself is not a problem in general but breaks in
+    # the address sanitizer tests
+    OPENSSL_CONF_ORIG=${OPENSSL_CONF}
+    export OPENSSL_CONF=/dev/null
+    pkcs11-tool "$@"
+    export OPENSSL_CONF=${OPENSSL_CONF_ORIG}
 }
```

Comment:
Did you try to run the tests without this wrapper once the `LD_PRELOAD` was correctly set? My assumption was that this could be the problem, but it might not be in the end and it could have been only red herring.

Reply:
Yes I did try with `$CHECKER pkcs11-tool "$@"` but it still failed

Reply:
Turns out with a simple addition to the suppression file I am able to use pkcs11-tool without the openssl.cnf switcheroo and just using the proper checker prefix